flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -552,6 +566,39 @@ public Table sqlQuery(String query) {
 		}
 	}
 
+	@Override
+	public TableResult executeSql(String statement) {
+		List<Operation> operations = parser.parse(statement);
+
+		if (operations.size() != 1) {
+			throw new TableException(UNSUPPORTED_QUERY_IN_EXECUTE_SQL_MSG);
+		}
+
+		Operation operation = operations.get(0);
+		if (operation instanceof CreateTableOperation ||
```

Comment:
duplicated check with the ones inside `executeOperation `

Reply:
we can simplify this